### PR TITLE
feat: introduce script to automate simple changes to components

### PR DESCRIPTION
**Pull request by Izook**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** # (issue, if applicable)

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [ ] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: scripts/config/propogateAuroChange.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# Script used to propogate a specific change to all of our repositories by making identical PRs into every repository.
+
+# Required:
+#   - GitHub CLI >= 2.31.0
+#   - AlaskaAirlines GitHub PAT
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply scripts/config/propogateAuroChange.sh:7]: This PAT. Is this unique for each user and only used in the CLI or does this have to be registered with every repo?

File: scripts/config/propogateAuroChange.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
```
Comment (blackfalcon) [inline scripts/config/propogateAuroChange.sh:1]: This document is a template, not actually useable in its current state? If that is the case, we should be more declarative about that.

File: scripts/config/propogateAuroChange.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+repos=$(gh repo list $repoOwner \
+        --source \
+        --visibility=public \
+        --json name \
+        --jq .[].name \
+      | grep "auro-" \
+      );
+
+# Clear Previous Workspace
+rm -rf ./workspace
```
Comment (blackfalcon) [inline scripts/config/propogateAuroChange.sh:52]: This feels like brute force. What if the repo is already using a `./workspace` dir?

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by blackfalcon** [comment]
There is no issue linking this work back to the Kanban. Temporarily assigned the PR to the project.

File: scripts/config/propogateAuroChange.sh [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  exit 1;
+fi
+if [ -z "$commitMessage" ]; then
+  echo "commitMessage undefined, please define using -c flag.";
+  exit 1;
+fi
+
+# Get All Auro Repositories
+repoOwner="AlaskaAirlines"
+repos=$(gh repo list $repoOwner \
```
Comment (Izook) [inline scripts/config/propogateAuroChange.sh:43]: We should update this filter to use an allow/block repo list that would empower us to apply this change to select repos or ignore certain repos.

**Comment by blackfalcon** [comment]
Closing this pull request and moving the issue to a Discussion for the team to review and discuss.

**Pull request closed without merging**